File: server/router/cloudpods/storage/snapshot.go
package storage

import (
	"github.com/gin-gonic/gin"
	v1 "ygpt/server/api/v1"
)

type StoSnapshotRouter struct {
}

func (s *StoSnapshotRouter) InitStoSnapshotRouter(Router *gin.RouterGroup) {
	dataRouter := Router.Group("cloud/storage/snapshot")
	vpcApi := v1.ApiGroupApp.CloudApiGroup.StorageApiGroup.StoSnapshotApi

	{
		dataRouter.POST("getSnapshotList", vpcApi.GetStoSnapshotList)      // 获取数据列表
		dataRouter.POST("snapshotReset", vpcApi.SnapshotReset)             // 回滚主机
		dataRouter.POST("deleteSnapshot", vpcApi.DeleteSnapshot)           // 删除主机快照
		dataRouter.POST("deleteSnapshotByIds", vpcApi.DeleteSnapshotByIds) // 删除主机快照
	}
}

File: web/src/api/cloudpods/storage/snapshot.js
import request from '@/utils/request'
//获取
export function getSnapshotList(params){
    return request({
        url:'cloud/storage/snapshot/getSnapshotList',
        method:'post',
        data: params
    })
}
//回滚主机
export function snapshotReset(params){
    return request({
        url:'cloud/storage/snapshot/snapshotReset',
        method:'post',
        data: params
    })
}
//删除主机快照
export function deleteSnapshot(params){
    return request({
        url:'cloud/storage/snapshot/deleteSnapshot',
        method:'post',
        data: params
    })
}
//批量删除主机快照
export function deleteSnapshotByIds(params){
    return request({
        url:'cloud/storage/snapshot/deleteSnapshotByIds',
        method:'post',
        data: params
    })
}